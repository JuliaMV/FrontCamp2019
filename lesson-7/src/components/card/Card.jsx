import React from 'react';
import PropTypes from 'prop-types';

import css from './Card.module.css';

const Card = ({
  img, title, genres, date,
}) => (
  <li className={css.card}>
    <img className={css.poster} src={img} alt="description" />
    <h3 className={css.title}>{title}</h3>
    <div className={css.meta}>
      <span className={css.genres}>{genres.join(' ')}</span>
      <span className={css.date}>{date}</span>
    </div>
  </li>
);

Card.propTypes = {
  title: PropTypes.string.isRequired,
  genres: PropTypes.arrayOf(PropTypes.string).isRequired,
  img: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
};

export default Card;
