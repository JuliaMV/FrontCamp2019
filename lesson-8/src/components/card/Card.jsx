import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import css from './Card.module.css';

const Card = ({
  img, title, genres, date, id,
}) => {
  return (
    <li className={css.card} key={id}>
      <Link to={`/film/${id}`}>
        <img className={css.poster} src={img} alt="description" />
      </Link>
      <div className={css.info}>
        <div className={css.meta}>
          <h3 className={css.title}>{title}</h3>
          <span className={css.genres}>{genres.join(' ')}</span>
        </div>
        <span className={css.date}>{new Date(date).getFullYear()}</span>
      </div>
    </li>
  );
};

Card.propTypes = {
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  genres: PropTypes.arrayOf(PropTypes.string).isRequired,
  img: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
};

export default Card;
