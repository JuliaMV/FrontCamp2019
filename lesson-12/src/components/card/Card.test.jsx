import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { Card } from './Card';

Enzyme.configure({ adapter: new Adapter() });


describe('Card component', () => {
  test('should render with props', () => {
    const props = {
      img: 'http://image.com',
      title: 'News title',
      genres: ['fantasy'],
      date: '03.02.2019',
      id: 1325,
    };
    // eslint-disable-next-line react/jsx-props-no-spreading
    const component = shallow(<Card {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
