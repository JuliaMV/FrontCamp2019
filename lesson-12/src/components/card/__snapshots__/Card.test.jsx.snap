// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render with props 1`] = `
<li
  key="1325"
>
  <Link
    to="/film/1325"
  >
    <img
      alt="description"
      src="http://image.com"
    />
  </Link>
  <div>
    <div>
      <h3>
        News title
      </h3>
      <span>
        fantasy
      </span>
    </div>
    <span>
      2019
    </span>
  </div>
</li>
`;
